### PR TITLE
Added S3 storage documentation in user guide

### PR DESCRIPTION
**Pull request by Hirevo**
This PR adds an entry in the user guide to document how to configure the newly-available S3 storage strategy, as well as how to authenticate the registry to S3.

### CODE REVIEW

**Comment by Hirevo** [comment]
@jgallagher-cs Since you are the one who implemented this feature, do you think I have missed something in these docs that might be of interest to users ?

File: docs/src/whats-available/crate-stores.md [hunk 1]
```diff
@@ -4,12 +4,43 @@ Available crate stores
...
+```toml
+region = ["custom", "https://my.custom.s3.endpoint/"]
+```
+
+### S3 Authentication
+
+In order to authenticate the registry to S3, you can either:
+
+- define both `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables when running the registry.
+- have credentials stored in `~/.aws/config` (called the 'profile') and giving the registry read permission to it.
```
Comment (jgallagher-cs) [inline docs/src/whats-available/crate-stores.md:44]: The [rusoto docs](https://docs.rs/rusoto_credential/0.44.0/rusoto_credential/struct.ChainProvider.html) for their default provider (which is what we use) list four options, checked in this order:

> 1. Environment variables: AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
> 2. credential_process command in the AWS config file, usually located at ~/.aws/config.
> 3. AWS credentials file. Usually located at ~/.aws/credentials.
> 4. IAM instance profile. Will only work if running on an EC2 instance with an instance profile/role.

I _think_ 2 and 4 are relatively rare compared to 1 and 3 (at least, I've personally used 1 and 3 but never 2 or 4). I'm not sure how much we want to duplicate their docs though. Maybe include a link to theirs? If we keep this more or less as-is, I think we should change to "have credentials stored in `~/.aws/credentials`".

**Comment by jgallagher-cs** [comment]
@Hirevo I left one minor comment on the docs for authentication; other than that, it looks good to me. Thanks! 👍

File: docs/src/whats-available/crate-stores.md (hunk 1 above)
Comment (Hirevo) [reply docs/src/whats-available/crate-stores.md]: I had included the two options that I used the most as well (1 and 3), but I think it should be ok to mention all 4 options in here, they're quite short to mention in full.  
I also included Rusoto's documentation link for more details if required.

**Comment by Hirevo** [comment]
@jgallagher-cs Thanks for the review !

Sorry for my long absence on this PR.  
This should be ready to go now.

**Pull request merged**